src/Facing.php

```diff
@@ -89,67 +88,58 @@ private function __construct(){
 	/**
 	 * Returns the axis of the given direction.
 	 */
-	public static function axis(int $direction) : int{
-		return $direction >> 1; //shift off positive/negative bit
+	public static function axis(Facing $direction) : Axis{
+		return match ($direction) {
+			self::DOWN, self::UP => Axis::Y,
+			self::NORTH, self::SOUTH => Axis::Z,
+			self::WEST, self::EAST => Axis::X,
+		};
 	}
 
 	/**
 	 * Returns whether the direction is facing the positive of its axis.
 	 */
-	public static function isPositive(int $direction) : bool{
-		return ($direction & self::FLAG_AXIS_POSITIVE) === self::FLAG_AXIS_POSITIVE;
+	public static function isPositive(Facing $direction) : bool{
+		return in_array($direction, [self::UP, self::SOUTH, self::EAST]);
 	}
 
 	/**
 	 * Returns the opposite Facing of the specified one.
-	 *
-	 * @param int $direction 0-5 one of the Facing::* constants
 	 */
-	public static function opposite(int $direction) : int{
-		return $direction ^ self::FLAG_AXIS_POSITIVE;
+	public static function opposite(Facing $direction) : Facing{
+		return match ($direction) {
+			self::DOWN => self::UP,
+			self::NORTH => self::SOUTH,
+			self::WEST => self::EAST,
+			self::UP => self::DOWN,
+			self::SOUTH => self::NORTH,
+			self::EAST => self::WEST,
```

Comment:
This should follow the order of the cases - it's a confusing mess right now

Reply:
How i can do ?